smithy-linters/src/main/java/software/amazon/smithy/linters/ShouldHaveUsedTimestampValidator.java

```diff
@@ -160,31 +172,43 @@ private List<ValidationEvent> validateUnion(
 
     private Stream<ValidationEvent> validateTargetShape(
             String name,
-            MemberShape target,
+            MemberShape memberShape,
             Model model,
             List<Pattern> patterns
     ) {
-        return OptionalUtils.stream(model.getShape(target.getTarget())
-                .flatMap(shape -> validateName(name, shape.getType(), target, patterns)));
+        return OptionalUtils.stream(model.getShape(memberShape.getTarget())
+                .flatMap(targetShape -> validateName(name, targetShape, memberShape, patterns, model)));
```

Comment:
I dont think the `patterns` value needs to be passed in as the validateName method is an instance method and patterns is an instance field.